homeassistant/components/climate/melissa.py

```diff
@@ -0,0 +1,248 @@
+"""
+Support for Melissa Climate A/C.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/climate.melissa/
+"""
+import logging
+
+from homeassistant.components.climate import ClimateDevice, \
+    SUPPORT_OPERATION_MODE, SUPPORT_TARGET_TEMPERATURE, SUPPORT_ON_OFF, \
+    STATE_AUTO, STATE_HEAT, STATE_COOL, STATE_DRY, STATE_FAN_ONLY, \
+    SUPPORT_FAN_MODE
+from homeassistant.components.fan import SPEED_LOW, SPEED_MEDIUM, SPEED_HIGH
+from homeassistant.components.melissa import DATA_MELISSA, DOMAIN
+from homeassistant.const import TEMP_CELSIUS, STATE_ON, STATE_OFF, \
+    STATE_UNKNOWN, STATE_IDLE, ATTR_TEMPERATURE, PRECISION_WHOLE
+
+DEPENDENCIES = [DOMAIN]
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_FLAGS = (SUPPORT_TARGET_TEMPERATURE | SUPPORT_OPERATION_MODE |
+                 SUPPORT_ON_OFF | SUPPORT_FAN_MODE)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Iterate through and add all Melissa devices."""
+    api = hass.data[DATA_MELISSA]
+    devices = api.fetch_devices().values()
+
+    all_devices = []
+
+    append = all_devices.append
+    for device in devices:
+        append(MelissaClimate(api, device['serial_number'], device))
+
+    add_devices(all_devices)
+
+
+class MelissaClimate(ClimateDevice):
+    """Representation of a Melissa Climate device."""
+
+    def __init__(self, api, serial_number, init_data):
+        """Initialize the climate device."""
+        self._name = init_data['name']
+        self._api = api
+        self._serial_number = serial_number
+        self._data = init_data['controller_log']
+        self._state = None
+        self._cur_settings = None
+
+    @property
+    def name(self):
+        """Return the name of the thermostat, if any."""
+        return self._name
+
+    @property
+    def is_on(self):
+        """Return current state."""
+        if self._cur_settings is not None:
+            return self._cur_settings[self._api.STATE] in (
+                self._api.STATE_ON, self._api.STATE_IDLE)
+        return STATE_UNKNOWN
```

Comment:
It could be useful to log the received state if it is not None, but isn't either STATE_ON nor STATE_IDLE. Not a biggie, but could be useful to debug the interaction if something goes awry at some later point.

Reply:
Yes you are correct, or actually that should be logged when fetching settings? This occurs if things blow up in ```self.update``` (or if hass just started)

Reply:
Hmm, I think it should be checked here in any case, as that's the "else" part and its good to program defensively even if that'd be checked earlier during update.